src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs

```diff
@@ -0,0 +1,187 @@
+                transferFeature.TransferRequest(wrapper);
+                return Task.FromResult(0);
+            });
+
+            var delegationProperty = delegator.Features.Get<IServerDelegationFeature>();
+            wrapper = delegationProperty.CreateDelegationRule(queueName, receiverAddress);
+
+            var responseString = await SendRequestAsync(delegatorAddress);
+            Assert.Equal(_expectedResponseString, responseString);
+            wrapper?.Dispose();
+        }
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task DelegateAfterWriteToResponseBodyShouldThrowTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+            using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
+            options =>
+            {
+                options.RequestQueueName = queueName;
+            });
+
+            DelegationRule wrapper = default;
+
+            using var delegator = Utilities.CreateHttpServer(out var delegatorAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+                var transferFeature = httpContext.Features.Get<IHttpSysRequestTransferFeature>();
+                Assert.Throws<InvalidOperationException>(() => transferFeature.TransferRequest(wrapper));
+            });
+
+            var delegationProperty = delegator.Features.Get<IServerDelegationFeature>();
+            wrapper = delegationProperty.CreateDelegationRule(queueName, receiverAddress);
+
+            var responseString = await SendRequestAsync(delegatorAddress);
+            Assert.Equal(_expectedResponseString, responseString);
+            wrapper?.Dispose();
+        }
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task WriteToBodyAfterDelegateShouldThrowTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+            using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
+            options =>
+            {
+                options.RequestQueueName = queueName;
+            });
+
+            DelegationRule wrapper = default;
+
+            using var delegator = Utilities.CreateHttpServer(out var delegatorAddress, async httpContext =>
+            {
+                var transferFeature = httpContext.Features.Get<IHttpSysRequestTransferFeature>();
+                transferFeature.TransferRequest(wrapper);
+                await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+                {
+                    await httpContext.Response.WriteAsync(_expectedResponseString);
```

Comment:
Are you sure this throws? I thought you just made writes no-op. With this testing pattern you wouldn't be able to observe an assert failure here after transferring the request.

Reply:
You're right. This doesn't throw.